x-pack/test/alerting_api_integration/spaces_only/tests/alerting/group4/builtin_alert_types/circuit_breaker/index_threshold_max_alerts.ts

```diff
@@ -157,26 +156,24 @@ export default function maxAlertsRuleTests({ getService }: FtrProviderContext) {
         });
       });
 
-      // because the "execute" event is written at the end of execution
-      // after getting the correct number of recovered-instance events, we're often not
-      // getting the final "execute" event. use the execution UUID to grab it directly
-
-      const recoveredEventExecutionUuid = recoveredEvents[0]?.kibana?.alert?.rule?.execution?.uuid;
-
+      // To avoid flakiness, fetch 10 execute events after recovered alerts and
+      // find the execute event right after the recovered event
       const finalExecuteEvents = await retry.try(async () => {
         return await getEventLog({
           getService,
           spaceId: Spaces.space1.id,
           type: 'alert',
           id: ruleId,
           provider: 'alerting',
-          filter: `kibana.alert.rule.execution.uuid:(${recoveredEventExecutionUuid})`,
-          actions: new Map([['execute', { gte: 1 }]]),
+          actions: new Map([['execute', { gte: 10 }]]),
```

Comment:
The rule seems to be running at a 6s interval, so before it would have waited ~6s, and with this it will take ~1m.  

Is there any way we could change this test to wait for the 17 EL `alert-recovered` messages instead?  I think that would then return as soon as their found, which I imagine might be less than 10 rule executions.

Reply:
Sure I can give it a shot